### ISSUE TITLE
[FEATURE] Smart Mastering - Pass expand-values as node* instead of string* for custom matching algorithm

### ISSUE THREAD

**Issue by sjiang99** [marklogic/marklogic-data-hub#4499]
For my project, there is a requirement to match driver’s license by matching the license number and the issuing state.

                <DriversLicense>
                                <driverLicenseNumber>N12345</driverLicenseNumber>
                                <driverLicenseState>CA</driverLicenseState>
                                <issueDate>2010-01-12</issueDate>
                </DriversLicense
 
The matcher needs to match the DL number and issuing state. I thought about creating a custom algorithm that defines “DriverLicense” as the matching property, and to be passed in to the algorithm as expand values.
However, currently the algorithm signature looks to be xs:string:
 
```
declare function algorithms:zip-match(
  $expand-values as xs:string*,
  $expand-xml as element(matcher:expand),
  $options-xml as element(matcher:options)
) as cts:query*
```
 
 
I tested out, it looks like base.xqy converts the property into string before calling the custom algorithm. So all the algorithm sees is a string of “N12345 CA 2010-01-12”. For the algorithm to work, the element itself being passed in instead.

**Comment by rjrudin** [comment]
Can this be closed, as it's something we expect to work OOTB in 5.4?

**Comment by fsnow** [comment]
This is covered by one of the scenarios on the smart mastering / structured properties wiki page for 5.4. I think it can be deleted.

**Comment by dmcassel** [comment]
Has this been tested on JSON data as well? I have a customer that would like to do matching based on a structure. When are you targeting a release that would include this?

**Comment by ryanjdew** [comment]
Closing due to this being shipped in the 5.4 release.
